post-processor/docker-push/post-processor_test.go

```diff
@@ -35,27 +35,39 @@ func TestPostProcessor_ImplementsPostProcessor(t *testing.T) {
 func TestGetEcrType(t *testing.T) {
 	awsConfig := docker.AwsAccessConfig{}
 
+	// Public
 	typ, _ := awsConfig.GetEcrType("https://public.ecr.aws/j9y7g6y8/dev_hc_pkr_dkr_test_1")
 	if typ != docker.Public {
 		msg := fmt.Sprintf("ECR type should be %v", docker.Public)
 		t.Fatal(msg)
 	}
 
-	typ, _ = awsConfig.GetEcrType("https://public.ecr.aws/j9y7g6y8/dev_hc_pkr_dkr_test_1")
+	typ, _ = awsConfig.GetEcrType("public.ecr.aws/j9y7g6y8/dev_hc_pkr_dkr_test_1")
 	if typ != docker.Public {
 		msg := fmt.Sprintf("ECR type should be %v", docker.Public)
 		t.Fatal(msg)
 	}
 
+	// Private
 	typ, _ = awsConfig.GetEcrType("https://12345.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1")
 	if typ != docker.Private {
 		msg := fmt.Sprintf("ECR type should be %v", docker.Private)
 		t.Fatal(msg)
 	}
 
+	typ, _ = awsConfig.GetEcrType("12345.dkr.ecr.us-east-1.amazonaws.com/private_dev_hc_pkr_dkr_test_1")
+	if typ != docker.Private {
+		msg := fmt.Sprintf("ECR type should be %v", docker.Private)
+		t.Fatal(msg)
+	}
+
+	// Known Unknown
 	typ, _ = awsConfig.GetEcrType("google.com")
-	if typ != docker.Invalid {
-		msg := fmt.Sprintf("ECR type should be %v", docker.Invalid)
+	if typ != docker.Private {
```

Comment:
I know the current code behaves like this now that we forcefully introduce a schema, and I'd imagine before the current change, if one specified `https://google.com` this would end-up in the `docker.Private` state, so this is probably out-of-scope for this PR, but is there any heuristic we can use to weed out easy invalids like this one?
Naively I'd think something like checking if the host is not the the shape of `\.dkr\.ecr\.[^.]+\.amazonaws\.com` we know it's not a private ECR URL?
Maybe I'm overthinking this, but I'd for sure prefer we reject more explicitely when the URI's not an ECR one, AFAICT there's basically two shapes for those so we can pinpoint them more accurately imho.

Reply:
Actually, before we introduced this code, we did not have any checks. We used to rely upon the AWS SDK. My thought process is to filter out ECR Public/Private URLs if we can otherwise let the AWS SDK handle the error raising as it used to be. And one more thing is that this part of the code is executed only for the AWS, so this is a minor edge case.

Reply:
I'm in the same light that we can probably do a bit more than force the HTTPS schema. It looks to me that the check for the protocol is not necessary at all, and that we should support URLs both with and without the schema. Is my observation correct? 

That said, it looks like you are using the GetEcrType to determine if an authorization token should be obtained for the Public ECR repository or for a Private repository. The difference being just the region. Is that correct? 

If so I think we can probably have logic that says if login server host is public.ecr.aws then we should get a PublicAuthorization token, for everything else we should treat as private. That said it is not clear to me if a user can CNAME the public gallery URL to something custom to the user, which in that case could be an issue for the auto detection. If CNAMing is an option should there be a configuration argument `public_ecr_gallery (bool)` that if true would get a Public authorization token? 


Lastly for the docker login command it looks like it does not matter if you pass it a login server with or without a protocol. So this should not be an issue unless I'm mistaken on the supported values.

Reply:
> Maybe I'm overthinking this, but I'd for sure prefer we reject more explicitely when the URI's not an ECR one, AFAICT there's basically two shapes for those so we can pinpoint them more accurately imho.

I don't you are over thinking here. Generally speaking we should try to validate for these values when possible. I don't know if it is as simple as checking for AWS owned domain if CNAMing is possible. Which is probably why the original code relied on the API to fail. This is not me saying we shouldn't it is more like we need to do research to make sure it does not run into the same issues as checking for a schema on the URL.